cli_tools/gce_image_publish/publish/publish.go

```diff
@@ -118,7 +118,7 @@ var (
 )
 
 // CreatePublish creates a publish object
-func CreatePublish(sourceVersion, publishVersion, workProject, publishProject, sourceGCS, sourceProject, ce, path string, varMap map[string]string, imagesCache map[string][]*computeAlpha.Image) (*Publish, error) {
+func CreatePublish(sourceVersion, publishVersion, workProject, publishProject, sourceGCS, sourceProject, ce, path, template string, varMap map[string]string, imagesCache map[string][]*computeAlpha.Image) (*Publish, error) {
```

Comment:
Being able to provide the same argument in two ways (a file path, and content) in the same function can be dangerous (what if both are provided?) and confusing. Having two separate functions (one can use the other), where one accepts content and another accepts file path is a cleaner way of achieving the same goal.

Reply:
I used your advice and moved this functionality into another function `CreatePublishWithTemplate` that calls `createPublish` but doesn't read in a file.